src/pylhe/__init__.py

```diff
@@ -120,14 +123,30 @@ def loads():
     pass
 
 
-def readLHEInit(thefile):
+def _open_gzip_file(filepath):
+    """
+    Checks to see if a file is compressed, and if so, open it with gzip
+    """
+    if not Path(filepath).name.lower().endswith(".gz"):
+        return filepath
+
+    with gzip.open(filepath, "r") as gzip_file:
+        try:
+            gzip_file.read(1)
```

Comment:
Need to _actually_ check that this is a `.gz` file here and not just that it is possible to read here, as a totally blank file can be read with `gzip_file.read(1)` just fine.

Reply:
I guess convention is fine.. we could check file headers.. but I feel like we could say that if you give pylhe a file endig w/ `.gz` it's your responsibility to ensure that it's  a gzipped file

Reply:
> we could check file headers

Yeah, I think this is a better way to do it.

Reply:
https://github.com/ahupp/python-magic this could be a library solution  or if we don'tt want tthe dependency read the fiirstt to bytes manually: https://stackoverflow.com/a/47080739/286382

Reply:
I had tried reading the first bytes manually last night, but I was getting errors for files produced by `mg5`, but I must have done something wrong as this works fine

```
$ file unweighted_events.lhe
unweighted_events.lhe: exported SGML document, ASCII text
$ file unweighted_events.lhe.gz 
unweighted_events.lhe.gz: gzip compressed data, was "tmp_0_unweighted_events.lhe", last modified: Tue Mar 30 07:22:36 2021, max compression, original size modulo 2^32 15580732
$ python
Python 3.8.6 (default, Jan  5 2021, 00:14:15) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> with open("unweighted_events.lhe.gz", "rb") as readfile:
...     print(readfile.read(2))
... 
b'\x1f\x8b'

```

Reply:
Oh it was because I was using `gzip.open` and not `open` to check:

```
>>> import gzip
>>> with gzip.open("unweighted_events.lhe.gz", "rb") as readfile:
...     print(readfile.read(2))
... 
b'<L'
```